db/migrate/20220617140542_move_become_method_from_options.rb

```diff
@@ -0,0 +1,51 @@
+class MoveBecomeMethodFromOptions < ActiveRecord::Migration[6.0]
+  ALLOWED_VALUES = %w[
+    sudo
+    su
+    pbrum
+    pfexec
+    doas
+    dzdo
+    pmrun
+    runas
+    enable
+    ksu
+    sesu
+    machinectl
+  ].freeze
+
+  class Authentication < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+    self.inheritance_column = :_type_disabled
+
+    serialize :options
+  end
+
+  def up
+    say_with_time("Move Authentication become_method out of options text field") do
+      Authentication.in_my_region.where("options LIKE '%become_method%'").find_each do |authentication|
+        next unless authentication.options&.key?(:become_method)
+
+        # Delete the old value out of options
+        become_method = authentication.options.delete(:become_method)
+
+        # Only set the new column if the existing value is valid
+        authentication.become_method = become_method if ALLOWED_VALUES.include?(become_method)
+
+        authentication.save!
+      end
+    end
+  end
+
+  def down
+    say_with_time("Move Authentication become_method back to options text field") do
+      Authentication.in_my_region.where.not(:become_method => nil).find_each do |authentication|
+        authentication.options = {} if authentication.options.nil?
+
+        authentication.options[:become_method] = authentication.become_method if ALLOWED_VALUES.include?(authentication.become_method)
+
+        authentication.save!
```

Comment:
Follow up comment to https://github.com/ManageIQ/manageiq-schema/pull/650/files#r900454286, I'm wondering if this should be:


```suggestion
        if ALLOWED_VALUES.include?(authentication.become_method)
          authentication.options ||= {}
          authentication.options[:become_method] = authentication.become_method
        end

        authentication.save!
```